### PR TITLE
feat: add GROUP BY, HAVING and Query/AdvancedQuery separation

### PR DESCRIPTION
**Pull request by vage88sg1**
Add support for GROUP BY and HAVING clauses with a clear separation between simple entity queries and projection queries.

API Changes:
- Query: simplified for entity-only queries (filter, sort, pagination, distinct)
- AdvancedQuery: for projections with SELECT, GROUP BY, HAVING clauses
- QueryDefinition: common interface for both query types
- Querity.advancedQuery(): new entry point for projection queries
- Querity.findAllProjected(): returns List<Map<String, Object>>

New Features:
- GROUP BY with property names or expressions
- HAVING clause for aggregate filtering
- AdvancedQueryPreprocessor for preprocessing advanced queries
- JpaQueryUtils to reduce code duplication

Implementation:
- JPA, MongoDB, Elasticsearch backend support
- Parser support for GROUP BY and HAVING
- Semantic validation (HAVING requires GROUP BY)
- Null validation in withPreprocessor() methods
- Immutable preprocessors list using List.copyOf()
- Standardized exception handling across all factories

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -298,6 +299,75 @@ Query query = QuerityParser.parseQuery(
...
 
+#### GROUP BY and HAVING
+
+Querity supports GROUP BY and HAVING clauses for aggregate queries. Use `groupBy` to group results and `having` to filter groups:
+
+```java
+import static io.github.queritylib.querity.api.Querity.*;
+
+// Group by single property
+Query query = Querity.query()
```
Comment (brunomendola) [inline README.md]: Shouldn't this be advancedQuery()?

Could you update all the README for the AdvancedQuery changes?

File: querity-spring-web/src/test/java/io/github/queritylib/querity/spring/web/QueritySpringWebTests.java [hunk 2]
```diff
@@ -36,9 +36,8 @@ class QueritySpringWebTests {
       /* not single condition */      "{\"filter\":{\"not\":{\"propertyName\":\"lastName\",\"operator\":\"EQUALS\",\"value\":\"Skywalker\"}}}",
       /* not conditions wrapper */    "{\"filter\":{\"not\":{\"and\":[{\"propertyName\":\"lastName\",\"operator\":\"EQUALS\",\"value\":\"Skywalker\"}]}}}",
       /* in array condition */        "{\"filter\":{\"propertyName\":\"lastName\",\"operator\":\"IN\",\"value\":[\"Skywalker\",\"Solo\"]}}",
-      /* simple select */             "{\"select\":{\"propertyNames\":[\"id\",\"firstName\",\"lastName\"]}}",
-      /* select with filter */        "{\"filter\":{\"propertyName\":\"lastName\",\"operator\":\"EQUALS\",\"value\":\"Skywalker\"},\"select\":{\"propertyNames\":[\"id\",\"name\"]}}",
-      /* select with sort */          "{\"select\":{\"propertyNames\":[\"id\"]},\"sort\":[{\"propertyName\":\"lastName\"}]}",
+      // Note: Tests with 'select' have been removed as Query no longer supports select.
+      // Use AdvancedQuery for projections (select, groupBy, having).
```
Comment (brunomendola) [inline querity-spring-web/src/test/java/io/github/queritylib/querity/spring/web/QueritySpringWebTests.java]: Could you add some mockMvc tests covering the AdvancedQuery cases?

I saw the DeserializerTests, but some edge cases could be found only when passing from a request.

**Review by brunomendola — CHANGES_REQUESTED** [review]
